cmd/celestia-appd/cmd/migrate_config.go

```diff
@@ -0,0 +1,214 @@
+	cfg := app.DefaultAppConfig()
+
+	v := viper.New()
+	v.SetConfigFile(configPath)
+	v.SetConfigType("toml")
+
+	if err := v.ReadInConfig(); err != nil {
+		return nil, fmt.Errorf("failed to read config file %s: %w", configPath, err)
+	}
+
+	if err := v.Unmarshal(cfg); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal config: %w", err)
+	}
+
+	return cfg, nil
+}
+
+// backupConfigFiles creates timestamped backups of the configuration files
+func backupConfigFiles(cometConfigPath, appConfigPath string) error {
+	timestamp := time.Now().Format("20060102-150405")
+
+	if err := backupFile(cometConfigPath, timestamp); err != nil {
+		return fmt.Errorf("failed to backup config.toml: %w", err)
+	}
+
+	if err := backupFile(appConfigPath, timestamp); err != nil {
+		return fmt.Errorf("failed to backup app.toml: %w", err)
+	}
+
+	fmt.Printf("Created backups with timestamp: %s\n", timestamp)
+	return nil
+}
+
+// backupFile creates a timestamped backup of a single file
+func backupFile(filePath, timestamp string) error {
+	if _, err := os.Stat(filePath); os.IsNotExist(err) {
+		fmt.Printf("Config file %s does not exist, skipping backup\n", filePath)
+		return nil
+	}
+
+	dir := filepath.Dir(filePath)
+	filename := filepath.Base(filePath)
+	ext := filepath.Ext(filename)
+	name := filename[:len(filename)-len(ext)]
+
+	backupPath := filepath.Join(dir, fmt.Sprintf("%s.%s.backup.%s", name, timestamp, ext))
+
+	data, err := os.ReadFile(filePath)
+	if err != nil {
+		return fmt.Errorf("failed to read file %s: %w", filePath, err)
+	}
+
+	if err := os.WriteFile(backupPath, data, 0o644); err != nil {
+		return fmt.Errorf("failed to write backup file %s: %w", backupPath, err)
+	}
+
+	fmt.Printf("Backed up %s to %s\n", filePath, backupPath)
+	return nil
+}
+
+// getSupportedVersions returns a list of supported updatable versions
+func getSupportedVersions() []string {
+	versions := make([]string, 0, len(updateRegistry))
+	for version := range updateRegistry {
+		versions = append(versions, version)
+	}
+	return versions
+}
+
+// applyV6Configs applies configuration changes needed for v6
+func applyV6Config(cmtCfg *config.Config, appCfg *serverconfig.Config) (*config.Config, *serverconfig.Config) {
+	fmt.Println("Applying v6 updates to configs...")
+
+	defaultCfg := app.DefaultConsensusConfig()
+	cmtCfg.RPC.MaxBodyBytes = defaultCfg.RPC.MaxBodyBytes
+
+	cmtCfg.Mempool.TTLNumBlocks = defaultCfg.Mempool.TTLNumBlocks
+	cmtCfg.Mempool.TTLDuration = defaultCfg.Mempool.TTLDuration
+	cmtCfg.Mempool.MaxTxBytes = defaultCfg.Mempool.MaxTxBytes
+	cmtCfg.Mempool.MaxTxsBytes = defaultCfg.Mempool.MaxTxsBytes
+	cmtCfg.Mempool.Type = defaultCfg.Mempool.Type
+	cmtCfg.Mempool.MaxGossipDelay = defaultCfg.Mempool.MaxGossipDelay
+
+	cmtCfg.P2P.SendRate = defaultCfg.P2P.SendRate
+	cmtCfg.P2P.RecvRate = defaultCfg.P2P.RecvRate
+
+	defaultAppCfg := app.DefaultAppConfig()
+	appCfg.MinGasPrices = defaultAppCfg.MinGasPrices
```

Comment:
My plan was to set this value to an empty string "" - to indicate that it wasn't set and to use the hardcoded default.

For now you can leave it as is and I will submit a PR on top of yours to make the right edits